Automattic/woocommerce-payments
# Add missing intent info when when stripe.intent webhook request is made

Fixes #4313 

#### Changes proposed in this Pull Request

In most cases, the intent info is updated during the UPE redirect, but sometimes Stripe webhook that delivers information about Stripe Intent can process faster than UPE redirect, which results in missing order metadata like `payment_method_id`. Since this webhook updates the status of the order, some metadata is skipped during the redirect. This PR solves that issue by updating the metadata that can be obtained from the webhook request. Every metadata value is checked and if the value doesn't exist, and it is not empty, the metadata will be stored. There is a function that can be used to update the metadata (`attach_intent_info_to_order`), but this function requires `customer_id` which doesn't exist inside webhook request, and some metadata values can be empty which will results in metadata with empty values.

#### Testing instructions

The easiest way on how to create an order and process it (finish payment) in a normal way. Go to the order and delete order metadata with keys like `_payment_method_id`, `_intent_id` or `_charge_id`. Go to the Stripe dashboard and rerun the `payment_intent.succeeded` webhook, and the metadata that you deleted should be there.

-------------------


**Post merge**
